indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceInputFormat.java

```diff
@@ -0,0 +1,61 @@
+package io.druid.indexer.hadoop;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+public class DatasourceInputFormat extends InputFormat<NullWritable, HadoopWritableInputRow>
+{
+
+  private static final Logger logger = LoggerFactory.getLogger(DatasourceInputFormat.class);
+
+  public static final String CONF_INPUT_SEGMENTS = "druid.segment.paths";
+  public static final String CONF_DRUID_SCHEMA = "druid.datasource.schema";
+
+  @Override
+  public List<InputSplit> getSplits(JobContext context) throws IOException, InterruptedException
```

Comment:
How would you feel about having a DataSegmentInputSplit as per https://github.com/druid-io/druid/pull/1351/files#diff-e3e429f5bc3e9ee67f25b69b7e6c3969R981

?


Alternatively we could have a "HadoopDataSegment" which has a hadoopy loadSpec which is only a URI parsable by hadoop (in addition to the other things a DataSegment usually has. That could allow the peon/overlord to setup stuff correctly for hadoop.

We have the data to get all the splits done completely, it is just a matter of ensuring the data gets propagated to Hadoop correctly.